src/main/java/seedu/address/model/person/UniqueScheduleList.java

```diff
@@ -0,0 +1,160 @@
+            throw new PersonNotFoundException();
+        }
+
+        if (!target.isSamePerson(editedPerson) && contains(editedPerson)) {
+            throw new DuplicatePersonException();
+        }
+        internalList.set(index, editedPerson);
+        updateInternalListRef();
+    }
+
+    /**
+     * Removes the equivalent person from the list.
+     * The person must exist in the list.
+     */
+    public void remove(Person toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new PersonNotFoundException();
+        }
+        updateInternalListRef();
+    }
+
+    public void setPersons(UniqueScheduleList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code persons}.
+     * {@code persons} must not contain duplicate persons.
+     */
+    public void setPersons(List<Person> persons) {
+        requireAllNonNull(persons);
+        if (!personsAreUnique(persons)) {
+            throw new DuplicatePersonException();
+        }
+
+        internalList.setAll(persons);
+    }
+
+    /**
+     *  Returns a {@code List<Person>} which contains the filtered person list.
+     */
+    private List<Person> getScheduleList() {
+        return internalList
+                .stream()
+                .filter(person -> LocalDate.now().equals(person.getAClass().date))
+                .sorted(Person::compareTo)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Update internalListRef
+     */
+    private void updateInternalListRef() {
+        List<Person> list = getScheduleList();
+        internalListRef.setAll(FXCollections.observableList(list));
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Person> asUnmodifiableObservableList() {
+        updateInternalListRef();
+        return internalUnmodifiableList;
+    }
+
+    @Override
+    public Iterator<Person> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof UniqueScheduleList // instanceof handles nulls
+                        && internalList.equals(((UniqueScheduleList) other).internalList));
+    }
+
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+
+    /**
+     * Returns true if {@code persons} contains only unique persons.
+     */
+    private boolean personsAreUnique(List<Person> persons) {
+        for (int i = 0; i < persons.size() - 1; i++) {
+            for (int j = i + 1; j < persons.size(); j++) {
+                if (persons.get(i).isSamePerson(persons.get(j))) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
```

Comment:
Top in 2040s cohort @Echomo-Xinyu would not agree with this O(n^2) solution.

Reply:
We are not graded for time complexity right? Since our application is only targeting about 50 students only.

Reply:
> We are not graded for time complexity right? Since our application is targeting about 50 students only.

Yes, I don't think time complexity is as important as readability in this module [source](https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#introduction-2). In this case, we can do it in a faster method by having a HashSet, but I feel it would be a bit unnecessary.

Reply:
But I think line 153 could be a violation of the Law of Demeter tho

Reply:
> But I think line 153 could be a violation of the Law of Demeter tho

Technically yes haha but I don't know whether util functions are considered, feels like they are not. One way is to create a static isEqual class in Person, then use Person.isEqual(person1, person2).

Reply:
@cadencjk @Echomo-Xinyu This method was actually copied over from UniquePersonList.java, so, I think that it should be fine